AppDevAnalytics/Classes/Analytics.swift

```diff
@@ -6,18 +6,30 @@
 //  Copyright © 2019 Kevin Chan. All rights reserved.
 //
 
-import Crashlytics
+import FirebaseAnalytics
 
-class AppDevAnalytics {
+public protocol Event {
+    var name: String { get }
+    var parameters: [String: Any]? { get }
+}
+
+public extension Event {
+    var parameters: [String: Any]? {
+        nil
+    }
+}
+
+public class AppDevAnalytics {
 
-    static let shared = AppDevAnalytics()
+    static public let shared = AppDevAnalytics()
 
     private init() {}
 
-    func log(_ payload: Payload) {
+    static public func logEvent(_ event: Event) {
```

Comment:
`static public func log(event: Event) {` is more idiomatic and concise. If it is clear from context, `log(_ event: Event)` would be even better.